### PR TITLE
OpenCensus Core Tracing Documentation

### PR DESCRIPTION
**Pull request by red2k18**

### CODE REVIEW

File: _docs/_latest/configure/instrumentation/index.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+## Configuring Instrumentation
+
+**Tracing**
+* Tracing is enabled by default and the traces are automatically 
+exported to `Stackdriver Trace`.
+* If you wish to disable tracing, you may do so by modifying the 
+`initialize_forseti_services.sh`script found under `install/gcp/scripts` to 
+remove `FORSETI_COMMAND+=" --enable tracing"` and redeploy Forseti. 
+* Tracing is done in a background thread.
```
Comment (blueandgold) [inline _docs/_latest/configure/instrumentation/index.md]: Do we have a page on how to use instrumentation?  Because lines 21-23 seems more appropriate for Use rather than Configuration.

File: _docs/_latest/configure/instrumentation/index.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+---
+
+## Configuring Instrumentation
+
+**Tracing**
+* Tracing is enabled by default and the traces are automatically 
+exported to `Stackdriver Trace`.
+* If you wish to disable tracing, you may do so by modifying the 
+`initialize_forseti_services.sh`script found under `install/gcp/scripts` to 
+remove `FORSETI_COMMAND+=" --enable tracing"` and redeploy Forseti.
```
Comment (blueandgold) [inline _docs/_latest/configure/instrumentation/index.md]: Hmmm, the way that `--enable-tracing` is flag is placed in the `install/gcp/scripts/initialize_forseti_services.sh`, it will not be configurable.  i.e meaning that everytime the VM is restarted it will pull the file from github again and write over any changes here.

If you really want to make it configurable, you would have to set this up in the `forseti_server_conf.yaml` file and the application would determine if it should run or not.

File: _docs/_latest/use/using-instrumentation.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# {{ page.title }}
+
+Opencensus tracing has been added to Forseti as part of instrumentation on 
+Forseti.
+
+---
+
+**Tracing**
+* Tracing is enabled by default and the traces are automatically exported to 
+`Stackdriver Trace`.
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]:
```
* Tracing is enabled by default and the traces are automatically exported to 
`Stackdriver Trace`, and viewable in GCP Console.
```

File: _docs/_latest/use/using-instrumentation.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+* Tracing is done in a background thread.
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]: I don't this fact is interesting for users.  We should remove.

File: _docs/_latest/use/using-instrumentation.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+* Traces are extremely helpful for latency analysis of gRPC calls.
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]: Can you be more descriptive here?  Can you give an example of how a user can make use of this, and walk them through?

File: _docs/_latest/use/using-instrumentation.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+* Stackdriver Exporter is the default exporter for traces.
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]: How is this relevant to people using this?  i.e. why does knowing about the exporter matter?  Is it configurable?

File: _docs/_latest/use/using-instrumentation.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+`Stackdriver Trace`, and viewable in GCP console.
+* Traces are extremely helpful for latency analysis of gRPC calls. Viewing 
+traces in GCP console will keep you informed about the time taken in ms to 
+create an inventory and/or perform various operations on the model. This data 
+is critical in analysing and debugging latency issues. 
+* Stackdriver Exporter is the default exporter for traces. Feel free to add in 
+support for the exporter of your choice as OpenCensus supports variety of 
+exporters such as Zipkin, Jaeger and contribute back. 
+
+
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]: Too many blank lines at the end.  We only have one blank line at the end of the file.

File: _docs/_latest/use/using-instrumentation.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+* Tracing is enabled by default and the traces are automatically exported to 
+`Stackdriver Trace`, and viewable in GCP console.
+* Traces are extremely helpful for latency analysis of gRPC calls. Viewing 
+traces in GCP console will keep you informed about the time taken in ms to 
+create an inventory and/or perform various operations on the model. This data 
+is critical in analysing and debugging latency issues. 
+* Stackdriver Exporter is the default exporter for traces. Feel free to add in 
+support for the exporter of your choice as OpenCensus supports variety of 
+exporters such as Zipkin, Jaeger and contribute back. 
+
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]: Should we add a `What's next` section, so that people can read more about OpenCensus.

image

File: _docs/_latest/use/using-instrumentation.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+**Tracing**
+* Tracing is enabled by default and the traces are automatically exported to 
+`Stackdriver Trace`, and viewable in GCP console.
+* Traces are extremely helpful for latency analysis of gRPC calls. Viewing 
+traces in GCP console will keep you informed about the time taken in ms to 
+create an inventory and/or perform various operations on the model. This data 
+is critical in analysing and debugging latency issues. 
+* Stackdriver Exporter is the default exporter for traces. Feel free to add in 
+support for the exporter of your choice as OpenCensus supports variety of
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]: How about we say this?
```
* Stackdriver Exporter is the default exporter for traces, but OpenCensus 
supports variety of exporters such as Zipkin and Jaeger. Feel free
to add support for the exporter of your choice, and please contribute back!
```

File: _docs/_latest/use/using-instrumentation.md [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+Opencensus tracing has been added to Forseti as part of instrumentation on 
+Forseti.
+
+---
+
+**Tracing**
+* Tracing is enabled by default and the traces are automatically exported to 
+`Stackdriver Trace`, and viewable in GCP console.
+* Traces are extremely helpful for latency analysis of gRPC calls. Viewing
```
Comment (blueandgold) [inline _docs/_latest/use/using-instrumentation.md]:
```
* Traces are extremely helpful for latency analysis of gRPC calls. For
example, viewing traces in GCP console will keep you informed
about the time taken for operations between the client and server to
complete, such as to create an inventory or data model. This is
critical in analyzing and debugging latency issues. 
```

File: _docs/_latest/use/using-instrumentation.md (hunk 9 above)
Comment (red2k18) [reply _docs/_latest/use/using-instrumentation.md]: Rephrased.

File: _docs/_latest/use/using-instrumentation.md (hunk 10 above)
Comment (red2k18) [reply _docs/_latest/use/using-instrumentation.md]: Done

File: _docs/_latest/use/using-instrumentation.md (hunk 7 above)
Comment (red2k18) [reply _docs/_latest/use/using-instrumentation.md]: Removed extra lines.

**Comment by dekuhn** [comment]
This PR has been open for a long time. Are you able to resolve the issues or can you close it?

File: _docs/_latest/use/using-instrumentation.md (hunk 8 above)
Comment (red2k18) [reply _docs/_latest/use/using-instrumentation.md]: Future work will be decided based on needs and users' request.

File: _docs/_latest/use/index.md [hunk 11]
```diff
@@ -19,3 +19,8 @@ Use the CLI to interface with Forseti and all of its modules.
 **[Get Help using Forseti]({% link _docs/latest/use/get-help.md %})**
 
 Find the best way to get help using Forseti.
+
+**[Using Instrumentation]
```
Comment (blueandgold) [inline _docs/_latest/use/index.md]: The dev setup needs to be documented, to enable tracing: such as the new API that needs to be enabled, optional packages that needs to be installed, flag that needs to be run.  Using tracing as part of the development flow would be awesome.

https://forsetisecurity.org/docs/latest/develop/dev/setup.html

File: _docs/_latest/develop/dev/setup.md [hunk 12]
```diff
@@ -242,6 +242,23 @@ cd <virtual environment>/lib/python2.7/site-packages/google/cloud
 ln -s <path to your git source code>/google/cloud/forseti forseti
 ```
 
+## Setting up instrumentation
```
Comment (blueandgold) [inline _docs/_latest/develop/dev/setup.md]: Did we also document somewhere, the new API that needs to be enabled?

File: _docs/_latest/use/index.md (hunk 11 above)
Comment (red2k18) [reply _docs/_latest/use/index.md]: Done

File: _docs/_latest/develop/dev/setup.md (hunk 12 above)
Comment (red2k18) [reply _docs/_latest/develop/dev/setup.md]: Documented in `using-instrumentation.md`.

**Comment by kevensen** [comment]
@red2k18 Is this PR being targeted for v2.20?

**Comment by blueandgold** [comment]
We will be using Cloud Profiler instead of Stackdriver Trace.
